core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/iterator/GroupIterator.java

```diff
@@ -654,16 +652,17 @@ private void operate(BindingSet bs, Predicate<?> predicate, Object t) {
 			var aggOperator = (AggregateFunctionCall) operator;
 			LongFunction<Predicate<Value>> predicate = operator.isDistinct() ? DistinctValues::new
 					: ALWAYS_TRUE_VALUE_SUPPLIER;
-			var factory = CustomAggregateFunctionRegistry.getInstance().get(aggOperator.getURI());
-			if (factory.isPresent()) {
-				var function = factory.get()
-						.buildFunction(new QueryStepEvaluator(strategy.precompile(aggOperator.getArg(), context)));
-				return new AggregatePredicateCollectorSupplier<>(
-						function,
-						predicate,
-						() -> factory.get().getCollector(),
-						ge.getName());
-			}
+			var factory = CustomAggregateFunctionRegistry.getInstance().get(aggOperator.getIRI());
+
+			var function = factory.orElseThrow(
+					() -> new QueryEvaluationException("Unknown aggregate function '" + aggOperator.getIRI() + "'"))
```

Comment:
Added an exception here, so that we match the behaviour of the custom function code.